src/main/kotlin/com/ampnet/projectservice/grpc/userservice/UserServiceImpl.kt

```diff
@@ -41,4 +42,20 @@ class UserServiceImpl(
             throw GrpcException(ErrorCode.INT_GRPC_USER, "Failed to fetch users. ${ex.localizedMessage}")
         }
     }
+
+    override fun getUsersByEmail(emails: List<String>): List<UserResponse> {
+        logger.debug { "Fetching users by emails: ${emails.joinToString()}" }
+        try {
+            val request = GetUsersByEmailRequest.newBuilder()
+                .addAllEmails(emails)
+                .build()
+            val response = serviceBlockingStub
+                .withDeadlineAfter(applicationProperties.grpc.userServiceTimeout, TimeUnit.MILLISECONDS)
```

Comment:
This is duplication and can be extacted, see `WalletServiceImpl.serviceWithTimeout`